### DISCUSSION TITLE
How to pass a custom mask to my loss function?

### DISCUSSION THREAD

**Discussion by intercodesys** [timeseriesAI/tsai#562] [Q&A]
I want to be able to use a mask in my loss function, so it focus on specific items more than others. I tried to define the mask and pass it to the loss function: loss_func = partial(my_loss, mask) which obviously does not work because the dataloader shuffles the items in every batch. 

So my question is how do I do this? Can I pass the mask to the dataloader somehow and retrieve it from the loss function? Or can I get the shuffled indices of the current batch from within the loss function?

Thanks!

**Comment by vrodriguezf** [comment]
I'd have a look at how it is done in the implementation of the Masked Value
Predictor callback.

**Reply by intercodesys** [reply]
Thank you for your suggestion and I enjoyed to read the MVP code, but it seems like their mask always is a function of x, so in that case they do not need to know the indices of the current shuffled batch. In my case the mask is static, and is not a function of x, so it doesn't help. I tried to use mymask[self.dls.idxs] to get mask of the current batch, but it doesn't work if the trainsize % bs != 0 (last batch idxs always has same size as the bs)

**Reply by vrodriguezf** [reply]
what about enabling `drop_last` in the two dataloaders to skip the last batch?
